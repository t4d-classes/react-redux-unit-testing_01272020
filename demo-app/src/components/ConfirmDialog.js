import React from 'react';

import styles from './ConfirmDialog.module.css';

export const ConfirmDialog = ({ message, onYes, onNo }) => {

  let msg = '';
  
  if (message) {
    msg = message;
  } else {
    msg = 'Are you sure?';
  }

  return <form className={styles.contentCenter}>
    <p>{msg}</p>
    <fieldset className={styles.noBorder}>
      <button type="button" onClick={onYes}>Yes</button>
      <button type="button" onClick={onNo} className={styles.noGap}>No</button>
    </fieldset>
  </form>

};