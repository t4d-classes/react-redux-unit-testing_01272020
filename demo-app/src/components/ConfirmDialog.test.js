import React from 'react';
import { render } from 'enzyme';

import { ConfirmDialog } from './ConfirmDialog';

describe('Confirm Dialog', () => {

  test("Confirm dialog default message", () => {
    const confirmDialogHtml = render(
      <ConfirmDialog onYes={() => null} onNo={() => null}  />
    ).html();
    const component = JSON.stringify(confirmDialogHtml);
    expect(component).toMatchSnapshot();
  });

  test("Confirm dialog custom message", () => {
    const confirmDialogHtml = render(
      <ConfirmDialog message="Do you want to delete this?" onYes={() => null} onNo={() => null}  />
    ).html();
    const component = JSON.stringify(confirmDialogHtml);
    expect(component).toMatchSnapshot();
  });

});