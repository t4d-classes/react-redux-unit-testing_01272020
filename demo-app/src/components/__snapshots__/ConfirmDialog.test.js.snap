// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirm Dialog Confirm dialog custom message 1`] = `"\\"<p>Do you want to delete this?</p><fieldset class=\\\\\\"noBorder\\\\\\"><button type=\\\\\\"button\\\\\\">Yes</button><button type=\\\\\\"button\\\\\\" class=\\\\\\"noGap\\\\\\">No</button></fieldset>\\""`;

exports[`Confirm Dialog Confirm dialog default message 1`] = `"\\"<p>Are you sure?</p><fieldset class=\\\\\\"noBorder\\\\\\"><button type=\\\\\\"button\\\\\\">Yes</button><button type=\\\\\\"button\\\\\\" class=\\\\\\"noGap\\\\\\">No</button></fieldset>\\""`;
